### PR TITLE
APEXMALHAR-2404  Avro file input operator should call beginWindow of AbstractFileInputOperator.

### PR DESCRIPTION
**Pull request by prasannapramod**
@devtagare   @sandeshh please see

### CODE REVIEW

**Comment by vrozov** [comment]
@prasannapramod Please fix PR title and commit comments. See contribution guidelines http://apex.apache.org/contributing.html

**Comment by prasannapramod** [comment]
@devtagare @sandeshh please see

File: contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java [hunk 1]
```diff
@@ -88,6 +88,10 @@
   @VisibleForTesting
   int fieldErrorCount = 0;
 
+  public AvroToPojo() {
```
Comment (devtagare) [inline contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java]: Why do we need the default constructor here ?

File: library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java [hunk 2]
```diff
@@ -907,38 +910,75 @@ public void testStateWithIdempotency() throws Exception
...
   {
-    FileContext.getLocalFSFileContext().delete(new Path(new File(testMeta.dir).getAbsolutePath()), true);
+    testIdempotencyWithCheckPoint(new LineByLineFileInputOperator(), new CollectorTestSink<String>(), new IdempotencyTestDriver<LineByLineFileInputOperator>()
+    {
+      @Override
+      public void writeFile(int count, String fileName) throws IOException
+      {
+        List<String> lines = Lists.newArrayList();
+        for (int line = 0; line < count; line++) {
+          //lines.add("f" + file + "l" + line);
```
Comment (devtagare) [inline library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java]: remove the comment

File: contrib/src/test/java/com/datatorrent/contrib/avro/AvroFileInputOperatorTest.java [hunk 3]
```diff
@@ -149,7 +152,52 @@ public void testSingleFileAvroReads() throws Exception
...
+          GenericRecord rec = new GenericData.Record(new Schema.Parser().parse(AVRO_SCHEMA));
+          rec.put("orderId", count * 1L);
+          rec.put("customerId", count * 2);
+          rec.put("total", count * 1.5);
+          rec.put("customerName", "*" + count + "*");
+          count--;
+          recordList.add(rec);
+        }
+        
+        //FileUtils.write(new File(testMeta.dir, fileName), org.apache.commons.lang3.StringUtils.join(recordList, '\n'));
```
Comment (devtagare) [inline contrib/src/test/java/com/datatorrent/contrib/avro/AvroFileInputOperatorTest.java]: please remove comments

File: library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java [hunk 4]
```diff
@@ -907,38 +910,75 @@ public void testStateWithIdempotency() throws Exception
...
 
     // empty file
     file = 2;
-    lines = Lists.newArrayList();
-    FileUtils.write(new File(testMeta.dir, "file" + file), StringUtils.join(lines, '\n'));
+    driver.writeFile(0, "file" + file);
 
 
-    LineByLineFileInputOperator oper = new LineByLineFileInputOperator();
+    //LineByLineFileInputOperator oper = new LineByLineFileInputOperator();
```
Comment (devtagare) [inline library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java]: remove comments please

File: contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java (hunk 1 above)
Comment (prasannapramod) [reply contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java]: Needed for de-searlization of operator during recovery.

**Comment by prasannapramod** [comment]
@devtagare @sandeshh addressed comments. Please see

File: library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java [hunk 5]
```diff
@@ -907,38 +910,72 @@ public void testStateWithIdempotency() throws Exception
...
+
+      @Override
+      public Context.OperatorContext getContext()
+      {
+        return testMeta.context;
+      }
+    });
+  }
+
+  public interface IdempotencyTestDriver<O extends Operator>
```
Comment (sandeshh) [inline library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java]: Prefer T over 0

File: library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java [hunk 6]
```diff
@@ -907,38 +910,72 @@ public void testStateWithIdempotency() throws Exception
...
+    void writeFile(int count, String fileName) throws IOException;
+
+    void setSink(O operator, Sink<?> sink);
+
+    String  getDirectory();
+
+    Context.OperatorContext getContext();
+  }
+
+  public static <O extends AbstractFileInputOperator, T> void testIdempotencyWithCheckPoint(O oper, CollectorTestSink<T> queryResults, IdempotencyTestDriver<O> driver) throws Exception
```
Comment (sandeshh) [inline library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java]: S over 0 for type

File: contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java [hunk 7]
```diff
@@ -61,7 +61,7 @@
 public class AvroFileInputOperator extends AbstractFileInputOperator<GenericRecord>
 {
 
-  private transient long offset = 0L;
+  protected transient long offset = 0L;
```
Comment (sandeshh) [inline contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java]: Why is this change required?

File: contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java (hunk 1 above)
Comment (tweise) [reply contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java]: In that case it can be private with a comment. You find many occurrences of the same elsewhere.

Comment (prasannapramod) [reply contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java]: This is not needed I thought the problem was that there was a non default constructor, that is not the case.

File: contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java (hunk 7 above)
Comment (prasannapramod) [reply contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java]: Changed not required. Reverting back.

**Comment by sandeshh** [comment]
test this please

**Comment by sandeshh** [comment]
test this please
